### PR TITLE
1965 credential endpoints

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Adds stolen and configured credentials endpoints

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?
* [ ] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by unit tests added
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/tests/unit_tests/monkey_island/cc/resources/credentials/test_configured_propagation_credentials.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    configured_propagation_credentials_url = get_url_for_resource(ConfiguredPropagationCredentials)
+
+    resp = flask_client.get(configured_propagation_credentials_url)
+
+    assert resp.status_code == 200
+    actual_propagation_credentials = json.loads(resp.data)
+    assert len(actual_propagation_credentials) == 2
+
+    # TODO: delete the removal of monkey_guid key when the serialization of credentials
+    del actual_propagation_credentials[0]["monkey_guid"]
```
Comment (ilija-lazoroski) [inline monkey/tests/unit_tests/monkey_island/cc/resources/credentials/test_configured_propagation_credentials.py:24]: Serialization is already done. I think you need to rebase on develop

File: monkey/monkey_island/cc/resources/credentials/configured_propagation_credentials.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+from flask import jsonify, request
+
+from common.credentials import Credentials
+from monkey_island.cc.repository import ICredentialsRepository
+from monkey_island.cc.resources.AbstractResource import AbstractResource
+
+logger = logging.getLogger(__name__)
+
+
+class ConfiguredPropagationCredentials(AbstractResource):
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/resources/credentials/configured_propagation_credentials.py:12]: Export these classes from the `__init__.py`

File: monkey/monkey_island/cc/resources/credentials/stolen_propagation_credentials.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+from flask import jsonify, request
+
+from common.credentials import Credentials
+from monkey_island.cc.repository import ICredentialsRepository
+from monkey_island.cc.resources.AbstractResource import AbstractResource
+
+logger = logging.getLogger(__name__)
+
+
+class StolenPropagationCredentials(AbstractResource):
```
Comment (ilija-lazoroski) [inline monkey/monkey_island/cc/resources/credentials/stolen_propagation_credentials.py:12]: Export it from `__init__.py`

File: monkey/monkey_island/cc/app.py [hunk 4]
```diff
@@ -185,6 +191,9 @@ def init_restful_endpoints(api: FlaskDIWrapper):
     api.add_resource(FileUpload)
 
     api.add_resource(PropagationCredentials)
+    api.add_resource(ConfiguredPropagationCredentials)
+    api.add_resource(StolenPropagationCredentials)
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/app.py:195]: Are these resources necessary? Can you just check `if request.url.endswith('/stolen-credentials')` and return different data?

File: monkey/monkey_island/cc/resources/credentials/stolen_propagation_credentials.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    urls = ["/api/propagation-credentials/stolen"]
+
+    def __init__(self, credentials_repository: ICredentialsRepository):
+        self._credentials_repository = credentials_repository
+
+    def get(self):
+        return jsonify(self._credentials_repository.get_stolen_credentials())
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/resources/credentials/stolen_propagation_credentials.py:19]: Does `jsonify()` work as expected? Does it properly handle the enums? Or do we need to do `jsonify([Credentials.to_json(c) for c in self._credentials_repository.get_stolen_credentials()])`

File: monkey/monkey_island/cc/resources/credentials/stolen_propagation_credentials.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+    def post(self):
+        credentials = Credentials.from_mapping(request.json)
+        self._credentials_repository.save_stolen_credentials(credentials)
+        return {}, 204
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/resources/credentials/stolen_propagation_credentials.py:24]: I was expecting this to accept an array, not a single object.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**